src/atomate2/abinit/jobs/base.py

```diff
@@ -0,0 +1,815 @@
+            return
+
+        # Link path to dest if dest link does not exist.
+        # else check that it points to the expected file.
+        logger.info("Linking path {} --> {}".format(source, dest))
+        if not os.path.exists(dest) or not strict:
+            if self.settings.COPY_DEPS:
+                shutil.copyfile(source, dest)
+            else:
+                os.symlink(source, dest)
+            return dest
+        else:
+            # check links but only if we haven't performed the restart.
+            # in this case, indeed we may have replaced the file pointer with the
+            # previous output file of the present job.
+            if (
+                not self.settings.COPY_DEPS
+                and os.path.realpath(dest) != source
+                # and not self.restart_info
+            ):
+                msg = "dest {} does not point to path {}".format(dest, source)
+                logger.error(msg)
+                raise InitializationError(msg)
+
+    def resolve_restart_deps(self, prev_dir):
+        """Resolve dependencies for a job that is restarted."""
+        # To be implemented for specific types of job (a relaxation is not restarted the same way
+        # an scf is, or a non scf, or ...)
+
+    @staticmethod
+    def set_logger():
+        """Set a logger for pymatgen.io.abinit and abipy."""
+        # TODO: what to do here ?
+        # log_handler = logging.FileHandler('abipy.log')
+        # log_handler.setFormatter(logging.Formatter(logging.BASIC_FORMAT))
+        # logging.getLogger('pymatgen.io.abinit').addHandler(log_handler)
+        # logging.getLogger('abipy').addHandler(log_handler)
+        # logging.getLogger('abiflows').addHandler(log_handler)
+
+    def set_workdir(self, workdir):
+        """Set up the working directory.
+
+        This also sets up and creates standard input, output and temporary directories, as well as
+        standard file names for input and output.
+        """
+        self.workdir = os.path.abspath(workdir)
+
+        # Files required for the execution.
+        self.input_file = File(os.path.join(self.workdir, INPUT_FILE_NAME))
+        self.output_file = File(os.path.join(self.workdir, OUTPUT_FILE_NAME))
+        self.log_file = File(os.path.join(self.workdir, LOG_FILE_NAME))
+        self.stderr_file = File(os.path.join(self.workdir, STDERR_FILE_NAME))
+
+        # This file is produce by Abinit if nprocs > 1 and MPI_ABORT.
+        self.mpiabort_file = File(os.path.join(self.workdir, MPIABORTFILE))
+
+        # Directories with input|output|temporary data.
+        self.indir = Directory(os.path.join(self.workdir, INDIR_NAME))
+        self.outdir = Directory(os.path.join(self.workdir, OUTDIR_NAME))
+        self.tmpdir = Directory(os.path.join(self.workdir, TMPDIR_NAME))
+
+        # Create dirs for input, output and tmp data.
+        self.indir.makedirs()
+        self.outdir.makedirs()
+        self.tmpdir.makedirs()
+
+    def get_abinit_input_set(
```

Comment:
Can you refactor this out, like the equivalent function in the vasp module: https://github.com/materialsproject/atomate2/blob/614ace0ee657759091b6e39296614b07b42140c8/src/atomate2/vasp/files.py#L144

Reply:
done